### PR TITLE
Connection details for MySQL docker container

### PR DESCRIPTION
**Pull request by devspacenine**
falls back to the original connection settings

### CODE REVIEW

**Comment by chrislbs** [comment]
Can we maybe add some type of configuration file for this that the app reads on startup? A common pattern I use for docker is to have a script that does a regex replace of a flat configuration file with environment variables. 

I'm mostly concerned with baking in the docker specific environment variables into the application. I realize we currently have the credentials/connections hardcoded into the app but if we're going to make this change now I'd rather we do it in a very generic way.

**Comment by devspacenine** [comment]
So I was going to make an argument for using those env variables created by linking containers because of the random hosts/ports that are generated, but then I found out that docker has deprecated the --link argument for a new [network setting](https://docs.docker.com/engine/reference/run/#/network-settings).

**Comment by devspacenine** [comment]
But yes, I agree with you. Configs are a must early on in a project. [This](https://www.npmjs.com/package/getconfig) is my favorite config management package.

**Comment by chrislbs** [comment]
Yeah, the main issue is that this doesn't necessarily have to be ran against a linked docker container. The db could be running anywhere so those environment variables won't necessarily be there. For now I'd say our config could just be a json file that we read on startup. Don't really need to bring anything in to just parse some simple db info. Is node capable of doing something like...

node app.js /path/to/config ?

**Comment by devspacenine** [comment]
No, but that's exactly what that getconfig package does. I'll make a commit that shows it in action.

**Comment by devspacenine** [comment]
Done

**Comment by devspacenine** [comment]
so getconfig just

File: config/default.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "db": {
+        "host": "mysql",
```
Comment (chrislbs) [inline config/default.json]: So the issue here is that this wouldn't work in my local environment. How can we make it so the dev can specify the values for this information and not have it commited to the repository?

I'd rather have something like config/example.json and then the consumer would specify the actual configuration and have it be gitignored.

Comment (devspacenine) [inline config/default.json]: You can make it load whichever config file you want with the NODE_ENV environment variable.
```bash
export NODE_ENV=chris
```
And that would try to load a config file at "config/chris.json".

Comment (devspacenine) [inline config/default.json]: Should probably put that in your .bashrc, though, if you want it to persist.

Comment (chrislbs) [inline config/default.json]: So the getConfig module expects a NODE_ENV and it reads a filename from a config/ directory from the CWD of where you ran 'node app' and then bases the filename off of the setting of NODE_ENV? And uses 'default' if no environment variable is set?

Comment (devspacenine) [inline config/default.json]: correct

Comment (chrislbs) [inline config/default.json]: Yeah, I'd say let's have 2 configurations saved. One is 'local' that assumes you have everything running locally on default ports. Another could be docker-dev or something that assumes you're using the docker images to run.

Comment (devspacenine) [inline config/default.json]: You don't think the default should be the docker config?

**Pull request merged**